src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -95,6 +95,13 @@ def validate(self, attrs):
         # project_rule(_details) endpoints by setting it on attrs
         actions = attrs.get("actions", tuple())
         for action in actions:
+            # XXX(colleen): For ticket rules we need to ensure the user has
+            # at least done minimal configuration
+            if action["id"] in TICKET_ACTIONS:
+                if not action.get("issuetype") or not action.get("project"):
```

Comment:
@ceorourke this isn't going to work for ADO which relies on `work_item_type`. I think we need some abstraction that will determine the additional required fields based on the value of `action["id"]`. Maybe we could add a field to the `JiraCreateTicketAction` and `AzureDevopsCreateTicketAction` classes like `additional_required_fields`. In the serializer, we'd just iterate through each of the required fields and ensure they are set.

Reply:
@scefali okay I updated it to just check for the existence of `dynamic_form_fields` which doesn't exist if they haven't clicked "Apply Changes" on the modal.